src/System.Resources.Reader/src/System/Resources/StreamExtensions.cs

```diff
@@ -78,13 +80,35 @@ public static string ReadString(this Stream stream)
                 int justRead = stream.Read(buffer, totalRead, stringLength - totalRead);
                 if(justRead == 0)
                 {
-                    throw new BadImageFormatException(SR.Resources_StreamNotValid);
+                    throw new EndOfStreamException(SR.BadImageFormat_ResourceNameCorrupted_NameIndex + position);
```

Comment:
Interesting, I just copy pasted it from line 120. The message is strange and has a misspelling. I would like to fix it, but it will change the semantics a bit.